### PR TITLE
add unittest for .read() and utf-8 #9024

### PR DESCRIPTION
**Pull request by manuel-sommer**
unittest to catch errors described in #9024 and #9017

### CODE REVIEW

**Comment by manuel-sommer** [comment]
This will fail at least for bundler

**Comment by manuel-sommer** [comment]
This is a first draft. Could you write your thoughts on this @kiblik ?

**Comment by manuel-sommer** [comment]
Waiting for the merge of:
https://github.com/DefectDojo/django-DefectDojo/pull/9018
https://github.com/DefectDojo/django-DefectDojo/pull/9025
Then, I will rebase

File: unittests/test_parsers.py
Comment (kiblik) [inline unittests/test_parsers.py:70]: Simple opening and reading of the file is definitely easier than running a subprocess with `cat`.

File: unittests/test_parsers.py
Comment (kiblik) [inline unittests/test_parsers.py:76]: This might be executed inside of `with self.subTest(parser=parser_dir.name, category='read utf-8'):`. Otherwise, evaluation would stop at this moment and other files would not be checked.

File: unittests/test_parsers.py
Comment (kiblik) [inline unittests/test_parsers.py:81]: In general, `read()` can be implemented also for other classes (not only for files). But I'm not sure, how to exclude these cases for now.

File: unittests/test_parsers.py
Comment (manuel-sommer) [reply unittests/test_parsers.py:66]: e.g. [sysdig_reports/sysdig_csv_parser.py:](https://github.com/DefectDojo/django-DefectDojo/blob/master/dojo/tools/sysdig_reports/sysdig_csv_parser.py)

File: unittests/test_parsers.py
Comment (manuel-sommer) [reply unittests/test_parsers.py:70]: This is what I tried. But, if you open a file with .read() or readlines() you get an Exception as soon as you hit a line which has .decode("utf-8") in it. This way was the only way I found to get back the content of each line.

File: unittests/test_parsers.py
Comment (manuel-sommer) [reply unittests/test_parsers.py:81]: I know, I will go through the failures once the mentioned PRs were merged and I did a rebase and add "utf-8" as a comment in a line below .read(). It will be something like:
# "utf-8" This comment was added to pass unittest  in file unittests/test_parsers.py. It solves the problem that unittests and production env output two different types: string and byte. In order to capture this, this check was implemented.

**Comment by kiblik** [comment]
@manuel-sommer, tried this and read of file works for me https://github.com/manuel-sommer/django-DefectDojo/pull/1

**Comment by manuel-sommer** [comment]
Lol, I don't know why it failed. :-D Now it works. Thank you @kiblik

**Comment by manuel-sommer** [comment]
This unittest proves that it would have caught https://github.com/DefectDojo/django-DefectDojo/pull/9018 and https://github.com/DefectDojo/django-DefectDojo/pull/9025

**Comment by manuel-sommer** [comment]
Ready to review @cneill @Maffooch @mtesauro @blakeaowens

**Review by mtesauro — APPROVED** [review]
Approve

**Review by grendel513 — APPROVED** [review]

**Review by cneill — APPROVED** [review]

**Review by blakeaowens — APPROVED** [review]

**Pull request merged**
